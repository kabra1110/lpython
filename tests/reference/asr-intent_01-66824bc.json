{
    "basename": "asr-intent_01-66824bc",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/intent_01.py",
    "infile_hash": "0a8fa2940567fccee2cfccd4af40f353b74dbe542590460a195246a5",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-intent_01-66824bc.stdout",
    "stdout_hash": "39edbdbd6bb1570dc829762e3103fac7b4bec3b2a408509f92b5e87f",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}