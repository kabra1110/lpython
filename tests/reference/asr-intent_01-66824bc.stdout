(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        6
                        {
                            Foo:
                                (StructType
                                    (SymbolTable
                                        2
                                        {
                                            p:
                                                (Variable
                                                    2
                                                    p
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    Foo
                                    []
                                    [p]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    [(())]
                                    ()
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            w:
                                                (Variable
                                                    3
                                                    w
                                                    []
                                                    Out
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Struct
                                                            6 Foo
                                                        )
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 5 (Integer 4)))]
                                                        PointerToDataArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    3
                                                    y
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            z:
                                                (Variable
                                                    3
                                                    z
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (List
                                                        (UnsignedInteger
                                                            4
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        [(Integer 4)
                                        (Real 8)
                                        (List
                                            (UnsignedInteger
                                                4
                                            )
                                        )
                                        (Array
                                            (Struct
                                                6 Foo
                                            )
                                            [((IntegerConstant 0 (Integer 4))
                                            (IntegerConstant 5 (Integer 4)))]
                                            PointerToDataArray
                                        )]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 3 x)
                                    (Var 3 y)
                                    (Var 3 z)
                                    (Var 3 w)]
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        5
                        {
                            
                        })
                    main_program
                    []
                    []
                )
        })
    []
)
